### PR TITLE
DIMACS exporter implementation

### PR DESCRIPTION
**Pull request by d-michail**
Added a simple DIMACS exporter to take care of issue #297.

### CODE REVIEW

File: jgrapht-ext/src/main/java/org/jgrapht/ext/DIMACSExporter.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import java.util.Objects;
+import java.util.Set;
+
+import org.jgrapht.Graph;
+
+/**
+ * Exports a graph into DIMACS format.
+ *
+ * <p>
+ * For a description of the format see <a href="http://dimacs.rutgers.edu/Challenges/">
```
Comment (jkinable) [inline jgrapht-ext/src/main/java/org/jgrapht/ext/DIMACSExporter.java]: Maybe I'm missing something, but I couldn't find a proper format description on this website? Which document did you use to implement this? I think in the past I looked at this: http://www.or.uni-bonn.de/lectures/ss12/praktikum/ccformat.pdf
for the graph coloring problem type.
Problem is, a lot of the urls on the page you reference simply don't work anymore :(

Either way, I would suggest to:
add a link to a page with a full format description
or
we can link to some pdfs. Optionally we can copy the pdf and put it in our website repository.

**Comment by d-michail** [comment]
I have also updated slightly the importer, to support all documented formats. This one is ready for merging when you find some time.

**Comment by jkinable** [comment]
Excellent, very nice to have full DIMACS support. Although I rarely import/export graphs, I really like this format since it's simple, concise and readable. 

I would be in favor of moving the import/export classes to their own package. jgrapht-io seems a good choice. I'm not sure how you can import jgrapht-io from jgrapht-ext. Wouldn't it suffice to move the import/export classes to the new package, and leave a deprecated copy of these classes in the jgrapht-ext package for 1 more release?

**Comment by jsichi** [comment]
Splitting off jgrapht-io makes sense.  I think @jkinable is correct that this requires copying the classes, but perhaps what @d-michail meant was that we should add a maven dependency from jgrapht-ext to jgrapht-io so that existing build files which reference jgrapht-ext won't have to be modified.

**Pull request merged**